src/Elmah/XmlFileErrorLog.cs

```diff
@@ -165,13 +165,37 @@ public override string Log(Error error)
                 writer.Flush();
             }
 
-            DeleteOldFiles();
+            // Do not try to delete files if the current exception was caused by a previous
+            // attempt to delete old files.
+            if (!IsXmlFileErrorLogDeleteException(error.Exception))
+            {
+                DeleteOldFiles();
+            }
+
             return errorId;
         }
 
         /// <summary>
+        /// Check if the exception or its InnerException is of type XmlFileErrorLogDeleteException.
+        /// InnerException is checked in in case the XmlFileErrorLogDeleteException is caught and wrapped
+        /// by some other code up the call stack.
+        /// </summary>
+        /// <param name="ex">Exception to test</param>
+        /// <returns>True if the exception or its InnerExcpetion if of type 
+        /// XmlFileErrorLogDeleteException, otherwise false.</returns>
+        private bool IsXmlFileErrorLogDeleteException(Exception ex)
+        {
+            if (ex == null) return false;
+
+            if (ex.GetType() == typeof(XmlFileErrorLogDeleteException)) return true;
```

Comment:
True, while `GetType()` is more specific, using `if (ex is XmlFileErrorLogDeleteException) return true;` would cover any subclasses, not that I expect many subclasses of this exception :)

Let me know if you want it changed.